src/FileSigChecker.cs

```diff
@@ -86,122 +82,131 @@ public static void ColorConsoleWriteLine(ConsoleColor color, object data)
-	    /// <returns>string</returns>
-		public static string GetHex(string fileName)
-		{
-		   	 string outHex = "";
-	   		 try
-		     {
-			        if (File.Exists(fileName))
-			        {
-			            byte[] getBytes = File.ReadAllBytes(fileName).Take(50).ToArray();
-			            outHex = HexDump.Hex(getBytes, 50);
-			        }
-		      }
-		      catch(Exception e)
-		      {
-		      		outHex = e.Message;
-		      }
-		      return outHex;
-		}
-	
-	    /// <summary>
-	    /// Hex dump
-	    /// </summary>
-	    /// <param name="bytes">Bytes input from file</param>
-	    /// <param name="bytesPerLine">Bytes per line. Default 16</param>
-	    /// <returns>string</returns>
-	    private static string Hex(byte[] bytes, int bytesPerLine = 50)
-	    {
-	        if (bytes == null) return "<null>";
-	        int bytesLength = bytes.Length;
-	
-	        char[] HexChars = "0123456789ABCDEF".ToCharArray();
-	
-	
-	        int firstCharColumn = 0
-	            + bytesPerLine * 3      
-	            + (bytesPerLine - 1) / 8;
-	            
-	
-	        int lineLength = firstCharColumn
-	            + bytesPerLine         
-	            + Environment.NewLine.Length;
-	
-	        char[] line = (new String(' ', lineLength - Environment.NewLine.Length) + Environment.NewLine).ToCharArray();
-	        int expectedLines = (bytesLength + bytesPerLine - 1) / bytesPerLine;
-	        StringBuilder result = new StringBuilder(expectedLines * lineLength);
-	
-	        for (int i = 0; i < bytesLength; i += bytesPerLine)
-	        {
-	
-	            int hexColumn = 0;
-	            int charColumn = firstCharColumn;
-	
-	            for (int j = 0; j < bytesPerLine; j++)
-	            {
-	                if (i + j >= bytesLength)
-	                {
-	                    line[hexColumn] = ' ';
-	                    line[hexColumn + 1] = ' ';
-	                }
-	                else
-	                {
-	                    byte b = bytes[i + j];
-	                    line[hexColumn] = HexChars[(b >> 4) & 0xF];
-	                    line[hexColumn + 1] = HexChars[b & 0xF];
-	                }
-	                hexColumn += 3;
-	            }
-	            result.Append(line);
-	        }
-	        return result.ToString();
-	    }
-	}
-}
+                        Console.WriteLine(outMessage);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    public class HexDump
+    {
+        /// <summary>
+        /// Hex dump
+        /// </summary>
+        /// <param name="fileName">File name for dump hex</param>
+        /// <returns>string</returns>
+        public static string GetHex(string fileName)
+        {
+            string outHex = "";
+            try
+            {
+                if (File.Exists(fileName))
+                {
+                    using (var stream = File.OpenRead(fileName))
+                    {
+                        int size = stream.Length < 50 ? (int)stream.Length : 50;
```

Comment:
will take 50 or if file is smaller than 50 just take the length of the file

Reply:
Good point